src/accordion/accordion.js

```diff
@@ -0,0 +1,110 @@
+// @flow
+import * as React from 'react';
+import {getOverrides} from '../helpers/overrides';
+import {Root as StyledRoot} from './styled-components';
+import {STATE_CHANGE_TYPE} from './constants';
+import type {
+  AccordionPropsT,
+  AccordionStateT,
+  StateChangeTypeT,
+  SharedStylePropsArgT,
+} from './types';
+
+export default class Accordion extends React.Component<
+  AccordionPropsT,
+  AccordionStateT,
+> {
+  static defaultProps = {
+    accordion: true,
+    disabled: false,
+    initialState: {
+      expanded: [],
+    },
+    onChange: () => {},
+    overrides: {},
+    stateReducer: (type, newState) => newState,
+  };
+
+  state = {
+    expanded: [],
+    ...this.props.initialState,
+  };
+
+  onPanelChange(key: string, onChange: () => {}, ...args: *) {
+    let activeKeys = this.state.expanded;
+    const {accordion} = this.props;
+    if (accordion) {
+      activeKeys = activeKeys[0] === key ? [] : [key];
+    } else {
+      activeKeys = [...activeKeys];
+      const index = activeKeys.indexOf(key);
+      const wasExpanded = index > -1;
+      if (wasExpanded) {
+        // remove active state
+        activeKeys.splice(index, 1);
+      } else {
+        activeKeys.push(key);
+      }
+    }
+    const newState = {expanded: activeKeys};
+    this.internalSetState(STATE_CHANGE_TYPE.expand, newState);
+    if (typeof onChange === 'function') onChange(...args);
+  }
+
+  internalSetState(type: StateChangeTypeT, changes: AccordionStateT) {
+    const {stateReducer, onChange} = this.props;
+    const newState = stateReducer(type, changes, this.state);
+    this.setState(newState);
+    onChange(newState);
+  }
+
+  getItems() {
+    const {expanded} = this.state;
+    const {accordion, disabled, children} = this.props;
+    const newChildren = [];
+
+    React.Children.forEach(children, (child, index) => {
```

Comment:
can we use `React.Children.map` here so we can get rid of the side effect of the iterator function?